### DISCUSSION TITLE
use c++ standard stream function in StrCat will be fail

### DISCUSSION THREAD

**Discussion by fansehep** [abseil/abseil-cpp#1248] [General]
If i use the StrCat in this case
```c++
std::string test () {
  return absl::StrCat(std::setw(sizeof(uint64_t)), std::setfill('0'), "some other things");
}
```
it will be fail. I want to know dose abseil-cpp provide similar function to control output?

**Comment by derekmauro** [comment]
The closest thing Abseil has to what I think you are trying to do is [`absl::StrFormat()`](https://abseil.io/docs/cpp/guides/format), but it requires that you use `printf`-style format strings. There is also `absl::StreamFormat()` if you want to use a format string in a stream.

If you are looking to use pure streams, standard C++ provides that.

**Reply by fansehep** [reply]
ok, Thanks, I have solve it by absl::StrFormat().
